packages/ios_platform_images/example/lib/main.dart

```diff
@@ -24,16 +24,47 @@ class _MyAppState extends State<MyApp> {
 
   @override
   Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final Color iconColor =
+        theme.iconTheme.color ?? theme.colorScheme.secondary;
+
     return MaterialApp(
       home: Scaffold(
         appBar: AppBar(
           title: const Text('Plugin example app'),
         ),
         body: Center(
           // "flutter" is a resource in Assets.xcassets.
-          child: Image(
-            image: IosPlatformImages.load('flutter'),
-            semanticLabel: 'Flutter logo',
+          child: Column(
+            children: <Widget>[
+              const Text('This is an icon from the iOS asset bundle'),
+              Image(
+                image: IosPlatformImages.load('flutter'),
+                semanticLabel: 'Flutter logo',
+              ),
+              const Text('These are icons from the iOS system'),
+              Image(
+                image: IosPlatformImages.loadSystemImage('face.smiling', 100,
+                    colors: <Color>[iconColor]),
+                semanticLabel: 'Smiling face',
+              ),
+              Image(
+                image: IosPlatformImages.loadSystemImage(
+                  'hare.fill',
+                  100,
+                  colors: <Color>[theme.colorScheme.secondary],
+                ),
+                semanticLabel: 'Sprinting hare',
+              ),
+              Image(
+                image: IosPlatformImages.loadSystemImage(
+                  'ladybug.fill',
+                  100,
+                  preferMulticolor: true,
+                ),
+                semanticLabel: 'Ladybug',
+              ),
+            ],
```

Comment:
I tried to implement this in [c14cd6e](https://github.com/flutter/plugins/pull/4803/commits/c14cd6ec6c549c192195ae40204264c4011022a2), the test passes but prints an error message after the fact. Not sure if that's acceptable, so I'm not marking this as resolved quite yet.

Reply:
Missed this, c14cd6e isn't linking to anything now.

Reply:
After rebasing, it's in 831cb53

Reply:
I have no idea how to test Flutter widgets, maybe @goderbauer has some more suggestions.  Ideally you could tell that it errored, but at least we know it didn't crash.  But that LGTM though.

Reply:
> the test passes but prints an error message after the fact.

What is printing the error message? If the error is expected, you should catch it in the text and expect it (e.g. with https://master-api.flutter.dev/flutter/flutter_test/WidgetTester/takeException.html). Running the test should not print anything to the console.

This may also be a case where a screenshot test may be useful to verify that the expected icon does get rendered (not sure if screenshot/golden tests are available in this repository).

Reply:
@goderbauer 

That was my initial approach, but I couldn't get it to work due to what I assume are some asynchrony issues on the iOS end.

I've tried reimplementing this approach locally with the following code:
```dart
  testWidgets(
    'ios system image error case',
    (WidgetTester tester) async {
      app.runErrorCaseApp();
      await tester.pumpAndSettle();

      expect(tester.takeException(), isArgumentError);
      expect(find.bySemanticsLabel('Nonexistant Symbol'), findsNothing);
    },
  );
```
`app.runErrorCaseApp()` calls `runApp` on the IOSExampleImageError class. https://github.com/flutter/plugins/blob/b23953a63b4294460e9e8fb0f40426e58019ac14/packages/ios_platform_images/example/lib/main.dart#L78-L90
However, despite throwing the argument error, the `takeException` method returns null. 
<details><summary>Test logs</summary>

```
❯ flutter test integration_test/ios_platform_images.dart
00:01 +0: ... /Users/caden/Developer/flutter/plugins/packages/ios_platform_images/example/integration_test/ios_platform_images.dart            Warning: Missing build name (CFBundleShortVersionString).
Warning: Missing build number (CFBundleVersion).
Action Required: You must set a build name and number in the pubspec.yaml file version field before submitting to the App Store.
00:03 +0: ... /Users/caden/Developer/flutter/plugins/packages/ios_platform_images/example/integration_test/ios_platform_images.dart            Ru00:04 +0: ... /Users/caden/Developer/flutter/plugins/packages/ios_platform_images/example/integration_test/ios_platform_images.dart        494ms
00:16 +0: ... /Users/caden/Developer/flutter/plugins/packages/ios_platform_images/example/integration_test/ios_platform_images.dart
00:19 +0: ... /Users/caden/Developer/flutter/plugins/packages/ios_platform_images/example/integration_test/ios_platform_images.dart      2,845ms
Xcode build done.                                           15.1s
00:22 +0: load ios bundled image
file:///Users/caden/Library/Developer/CoreSimulator/Devices/7674071D-43B8-44F4-AF74-D37EFE3999FB/data/Containers/Bundle/Application/E54E0078-C213-4CCD-A250-25CB8D054FDB/Runner.app/textfile
00:22 +1: load ios bundled image
file:///Users/caden/Library/Developer/CoreSimulator/Devices/7674071D-43B8-44F4-AF74-D37EFE3999FB/data/Containers/Bundle/Application/E54E0078-C213-4CCD-A250-25CB8D054FDB/Runner.app/textfile
00:22 +2: load ios bundled image
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following ArgumentError was thrown running a test (but after the test had completed):
Invalid argument(s): System image couldn't be found: nonexistent.sfsymbol

When the exception was thrown, this was the stack:
#2      IosPlatformImages.loadSystemImage.<anonymous closure> (package:ios_platform_images/ios_platform_images.dart:210:24)
#10     IosPlatformImages.loadSystemImage (package:ios_platform_images/ios_platform_images.dart:208:14)
#11     IOSImageExampleError.build (package:ios_platform_images_example/main.dart:88:36)
#12     StatelessElement.build (package:flutter/src/widgets/framework.dart:4827:28)
#13     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4754:15)
#14     Element.rebuild (package:flutter/src/widgets/framework.dart:4477:5)
#15     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4735:5)
#16     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4729:5)
#17     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3790:14)
#18     Element.updateChild (package:flutter/src/widgets/framework.dart:3524:20)
#19     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6291:14)
#20     Element.updateChild (package:flutter/src/widgets/framework.dart:3501:15)
#21     RenderObjectToWidgetElement._rebuild (package:flutter/src/widgets/binding.dart:1198:16)
#22     RenderObjectToWidgetElement.update (package:flutter/src/widgets/binding.dart:1175:5)
#23     RenderObjectToWidgetElement.performRebuild (package:flutter/src/widgets/binding.dart:1189:7)
#24     Element.rebuild (package:flutter/src/widgets/framework.dart:4477:5)
#25     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2659:19)
#26     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:882:21)
#27     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:363:5)
#28     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1144:15)
#29     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1081:9)
#30     LiveTestWidgetsFlutterBinding.handleDrawFrame (package:flutter_test/src/binding.dart:1458:13)
#31     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:995:5)
#35     _invoke (dart:ui/hooks.dart:151:10)
#36     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#37     _drawFrame (dart:ui/hooks.dart:115:31)
(elided 11 frames from dart:async and package:stack_trace)
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following ArgumentError was thrown running a test (but after the test had completed):
Invalid argument(s): System image couldn't be found: nonexistent.sfsymbol

When the exception was thrown, this was the stack:
#2      IosPlatformImages.loadSystemImage.<anonymous closure> (package:ios_platform_images/ios_platform_images.dart:213:24)
#10     IosPlatformImages.loadSystemImage (package:ios_platform_images/ios_platform_images.dart:208:14)
#11     IOSImageExampleError.build (package:ios_platform_images_example/main.dart:88:36)
#12     StatelessElement.build (package:flutter/src/widgets/framework.dart:4827:28)
#13     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4754:15)
#14     Element.rebuild (package:flutter/src/widgets/framework.dart:4477:5)
#15     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4735:5)
#16     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4729:5)
#17     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3790:14)
#18     Element.updateChild (package:flutter/src/widgets/framework.dart:3524:20)
#19     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6291:14)
#20     Element.updateChild (package:flutter/src/widgets/framework.dart:3501:15)
#21     RenderObjectToWidgetElement._rebuild (package:flutter/src/widgets/binding.dart:1198:16)
#22     RenderObjectToWidgetElement.update (package:flutter/src/widgets/binding.dart:1175:5)
#23     RenderObjectToWidgetElement.performRebuild (package:flutter/src/widgets/binding.dart:1189:7)
#24     Element.rebuild (package:flutter/src/widgets/framework.dart:4477:5)
#25     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2659:19)
#26     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:882:21)
#27     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:363:5)
#28     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1144:15)
#29     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1081:9)
#30     LiveTestWidgetsFlutterBinding.handleDrawFrame (package:flutter_test/src/binding.dart:1458:13)
#31     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:995:5)
#35     _invoke (dart:ui/hooks.dart:151:10)
#36     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#37     _drawFrame (dart:ui/hooks.dart:115:31)
(elided 11 frames from dart:async and package:stack_trace)
════════════════════════════════════════════════════════════════════════════════════════════════════
00:22 +2: ios system image error case
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: <Instance of 'ArgumentError'>
  Actual: <null>
   Which: is not an instance of 'ArgumentError'

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///Users/caden/Developer/flutter/plugins/packages/ios_platform_images/example/integration_test/ios_platform_images.dart:41:7)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Users/caden/Developer/flutter/plugins/packages/ios_platform_images/example/integration_test/ios_platform_images.dart line 41
The test description was:
  ios system image error case
════════════════════════════════════════════════════════════════════════════════════════════════════
00:22 +2 -1: ios system image error case [E]
  Test failed. See exception logs above.
  The test description was: ios system image error case
```
</details>